packages/react/src/components/header/Header.module.scss

```diff
@@ -50,10 +62,26 @@
     --logo-height: 32px;
   }
 
+  &.darkTheme {
```

Comment:
I think `.theme-dark` would be better as it's consistent with other themed classNames in HDS. If you search the project for `.theme` you'll see what I mean. There's just one odd ball `.dark` in Table scss.